app/src/main/java/com/travelguide/activities/TravelGuideActivity.java

```diff
@@ -159,7 +177,30 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
     public void selectDrawerItem(MenuItem menuItem) {
         switch (menuItem.getItemId()) {
             case R.id.login_fragment:
-                new LoginFragment().show(getSupportFragmentManager(), "Login_with_Facebook");
+                // new LoginFragment().show(getSupportFragmentManager(), "Login_with_Facebook");
+                // Login status also available in shared preference, but doing it using listener here.
+                // This is useful for later comparison.
+                LoginFragment.newInstance(new LoginFragment.OnLoginLogoutListener() {
```

Comment:
This is not right, take a look and see how was implemented on `TripListFragment`, you have to make `TravelGuideActivity` implement `OnLoginLogoutListener` and inside `LoginFragment` use attach and detach method properly, that way you don't need make it `Parcelable`.